### DISCUSSION TITLE
Enzyme shallow rendering support

### DISCUSSION THREAD

**Discussion by alexiusp** [palantir/blueprint#5884] [Q&A] [answered]
#### Environment

- __Package version(s)__: 3.15.0

#### Question

When I'm trying to unit-test my components with shallow rendering function of Enzyme I can not test for Blueprint component name, because they all look like 'Blueprint3.{name}' and enzyme find() function consider this selector as element with name 'Blueprint3' and a class 'Spinner'. So I need to set additional properties/class names even for simple Spinner only to be able to test if it's present in my layout or not.
What would you recommend in this situation? I do not want to use the mount rendering function when i just need to know if proper component was rendered or not.

**Comment by adidahiya** [comment, marked as answer]
You can import the component and query for it with Enzyme.

```ts
import { Button } from "@blueprintjs/core";

const wrapper = shallow(
	<Button text="hello" />
);
const button = wrapper.find(Button);
```

Note that this returns the React component node and not the `<button>` element; for the latter you will need to access `.hostNodes()`.
